pkg/core/interop/contract/call.go

```diff
@@ -36,23 +52,29 @@ func callInternal(ic *interop.Context, h []byte, name string, f callflag.CallFla
 	if err != nil {
 		return fmt.Errorf("contract not found: %w", err)
 	}
-	if strings.HasPrefix(name, "_") {
+	if strings.HasPrefix(method, "_") {
 		return errors.New("invalid method name (starts with '_')")
 	}
-	md := cs.Manifest.ABI.GetMethod(name)
+	md := cs.Manifest.ABI.GetMethod(method)
 	if md == nil {
 		return errors.New("method not found")
 	}
 	hasReturn := md.ReturnType != smartcontract.VoidType
 	if !hasReturn {
 		ic.VM.Estack().PushVal(stackitem.Null{})
 	}
+	return callInternal(ic, cs, method, fs, hasReturn, args)
+}
+
+func callInternal(ic *interop.Context, cs *state.Contract, name string, f callflag.CallFlag,
+	hasReturn bool, args []stackitem.Item) error {
+	md := cs.Manifest.ABI.GetMethod(name)
 	if md.Safe {
 		f &^= callflag.WriteStates
 	} else if ctx := ic.VM.Context(); ctx != nil && ctx.IsDeployed() {
```

Comment:
`IsDeployed` is probably `NEF != nil` now.

Reply:
If `NEF` is replaced with `Tokens` we can do it, but with some protection (ensure than `Tokens` will never be nil in a valid NEF struct.

Reply:
I've tried this, turns out sometimes `Tokens` are nil. Left a `NEF`, replaced `IsDeployed`.